app/Repositories/Wings/DaemonFileRepository.php

```diff
@@ -35,7 +35,9 @@ public function getContent(string $path, int $notLargerThan = null): string
             throw new DaemonConnectionException($exception);
         }
 
-        $length = (int) $response->getHeader('Content-Length')[0] ?? 0;
+        $fileLength = $response->getHeader('Content-Length');
+        
+        $length = (int) $fileLength ? $fileLength[0] : 0;
```

Comment:
The exception is because the cast has a higher precedence than the null coalesce. This is what we want:

```suggestion
        $length = (int) ($response->getHeader('Content-Length')[0] ?? 0);
```